### ISSUE TITLE
Add a route that presents both options (DNS & text file)

### ISSUE THREAD

**Issue by mrose17** [brave-intl/publishers#50]
here is what the publisher sees after doing the email verification:
![0](https://cloud.githubusercontent.com/assets/400552/23190709/65d9790e-f84e-11e6-95be-05a93726752b.png)
which is nice, but we need to do a bit of fine tuning.

first, we should look at the SOA for the publisher domain to see who the DNS provider is. if we have an entry for the provider in the list of "easy to use DNS provider sites", then the text above should be "flipped" to ask them to try the DNS approach, and the big orange button should take them to a page with the current overview PLUS the provider-specific instructions.

second, if they can't figure out how to do it, then they should get a button that does a customized `mailto:` to zendesk.

**Comment by alexwykoff** [comment]
aw : seems like creating the .well-known/text file would make sense 

mr : does this need more spec/sprucing?

bc : we can expose routes and show users the options

mr : is the wording sufficent ?

bc : explain what needs to be displayed on the screen

to summarize : we need specs to clarify the layout and copy @bradleyrichter @alexwykoff

**Comment by bsclifton** [comment]
Here's the route for the text file option:
https://github.com/brave/publishers/blob/6fb3377aa1f8334f0897acb98d3cd49de0f1f776/config/routes.rb#L15

**Comment by mrose17** [comment]
@bradleyrichter - before we can proceed on this, we need to review the wording.

@dgeb & @lgebhardt - when we start on this issue, the first thing is to restore that route and then @bradleyrichter can figure out what other changes we need to make...

**Comment by alexwykoff** [comment]
Chatted with Brad and he mentioned a missing 3rd option which is the wordpress extension made by Marshall.

**Comment by alexwykoff** [comment]
Following up on this, perhaps worthy of a separate issue, we should add detection on domain to know if they're using wordpress and offer the option conditionally.

**Comment by alexwykoff** [comment]
Here are a few updated ideas. I'll try to present them in order as they would appear in a flow:

1. Updated landing page :

In this example, we're looking to do a blue background with a fair amount of transparency, background image tbd. User options on the right.

Most notably, the shift to requiring an email upfront for validation for new, unverified publishers.

<img width="962" alt="screen shot 2017-09-06 at 8 56 03 am" src="https://user-images.githubusercontent.com/1434040/30121732-486cd5fe-92e1-11e7-85ab-f73ed56259c7.png">

2. Notification of verification email sent : 

We'd like to let the user know that they need to check their email to verify their address. We make sure to display the address given so they can notice if they made an error in their original input. (Envelope & cursor image is just a placeholder)

<img width="964" alt="screen shot 2017-09-06 at 9 02 37 am" src="https://user-images.githubusercontent.com/1434040/30122083-39a2534a-92e2-11e7-8093-373f9bc4b208.png">

3. Input of contact information : 

We'll simply ask for their name and domain they would like to verify. Phone number is completely optional.

<img width="960" alt="screen shot 2017-09-06 at 9 19 54 am" src="https://user-images.githubusercontent.com/1434040/30122783-9729384c-92e4-11e7-99fc-b24fef7c6e3b.png">


4. Detection or Choice : 

This will have some backend checks for either if the domain is wordpress hosted or if the domain contains *.github.io . If so, it will display instructions particular to those scenarios, otherwise it will show the users a choice.

4.a - Initial choice : 

Users will be presented with a choice between using a text verification or DNS. (Images in mock are placeholders)

<img width="962" alt="screen shot 2017-09-06 at 9 14 51 am" src="https://user-images.githubusercontent.com/1434040/30122833-b3dda6e4-92e4-11e7-914a-9a5ecc25560d.png">

4.b.i. - Text verification : 

If the domain gives a response via HTTPS, the user may proceed.

<img width="963" alt="screen shot 2017-09-06 at 9 22 30 am" src="https://user-images.githubusercontent.com/1434040/30122953-361671e0-92e5-11e7-8996-5ad58c5fe175.png">

4.b.ii. Missing HTTPS : 

If the domain does not give an HTTPS response, the user should be given a warning.

<img width="962" alt="screen shot 2017-09-06 at 9 25 30 am" src="https://user-images.githubusercontent.com/1434040/30122994-5c595958-92e5-11e7-95e5-ee6e0efe2be8.png">

If the domain then starts responding to HTTPS, the user is returned to 4.b.i.

4.c. DNS option : 

If the user chooses DNS they are shown the following.

<img width="962" alt="screen shot 2017-09-06 at 9 26 51 am" src="https://user-images.githubusercontent.com/1434040/30123116-b9f71546-92e5-11e7-8e2b-ea5144025ab5.png">

4. d. Wordpress option : 

If wordpress hosting is detected, they are given this option instead of having to choose like in 4.a.

<img width="961" alt="screen shot 2017-09-06 at 9 29 29 am" src="https://user-images.githubusercontent.com/1434040/30123158-ea947acc-92e5-11e7-9b07-0c8a2ffbd25f.png">

4.e. Github.io option : 

If *.github.io is used as the domain, they will be given an option similar to the text option from 4.a
(NOTE, this screen is work-in-progress)

<img width="963" alt="screen shot 2017-09-06 at 9 30 30 am" src="https://user-images.githubusercontent.com/1434040/30123226-2537eace-92e6-11e7-8983-6733ef1a848a.png">

4.f. Support queue : 

Aside from choosing text or DNS, a user may also choose to enter a support queue. If so, they will be shown the following.

<img width="961" alt="screen shot 2017-09-06 at 9 31 41 am" src="https://user-images.githubusercontent.com/1434040/30123301-6253d580-92e6-11e7-9496-b2e037700e11.png">

5. Notification of verification in progress : 

We will show the users that the verification in happening and they should check their email.

<img width="961" alt="screen shot 2017-09-06 at 9 33 31 am" src="https://user-images.githubusercontent.com/1434040/30123389-ae0ef9be-92e6-11e7-9d1f-5d389c56c296.png">

6.a. Verified domain, not on exclusion list : 

If the domain is verified, and listed a publisher (that is not on our exclusion list) they will be shown the following. 'Create BAT Wallet' link should go to the proper Uphold landing.

<img width="960" alt="screen shot 2017-09-06 at 9 36 36 am" src="https://user-images.githubusercontent.com/1434040/30123504-13c7e43c-92e7-11e7-8c86-d4db974d3f37.png">

6.b. Verified domain, on exclusion list : 

If the domain is verified, but it is listed as not a publisher (that is, it is on our exclusion list) they are shown the following. 

<img width="963" alt="screen shot 2017-09-06 at 9 42 57 am" src="https://user-images.githubusercontent.com/1434040/30123746-da7ee8a0-92e7-11e7-83da-d4d4c4d6947d.png">

6.b.i. Contention flow : 

Users will be given the opportunity to contest their status. 

<img width="962" alt="screen shot 2017-09-06 at 9 45 02 am" src="https://user-images.githubusercontent.com/1434040/30123829-16ebcd8a-92e8-11e7-9e12-c192e9f55801.png">

6.b.ii Contention flow part 2 : 

We will display a status so users will know their status is 'Under Review'.

<img width="962" alt="screen shot 2017-09-06 at 9 46 16 am" src="https://user-images.githubusercontent.com/1434040/30123881-40e9ad64-92e8-11e7-9060-a4aa64f0ccaf.png">


There will be a few updates coming, in particular on 4.b.i and 4.e.

**Comment by lgebhardt** [comment]
@alexwykoff A few questions, based on the steps:

4) Can you provide some details on the "backend check" that will be performed on the domain? Which service is performing the check and what are the return values?

4.b.i. & 4.b.ii) Is the HTTPS response also being checked by the backend check described in step 4? Or is this part of Publishers?

**Comment by lgebhardt** [comment]
From the developer documentation uphold provides their preferred images. Step 6a is using an Uphold logo which is not on the preferred list. Which Uphold logo file should this page use?

**Comment by alexwykoff** [comment]
@lgebhardt @mrose17 - for the .well-known method to work, we will need a check to see if the domain host is using valid HTTPS (no bad cert, etc. ) 
I would assume this should be have similarly to the validation check after the user has posted the .well-known validation file. I am unsure where or how these checks would be managed.

@lgebhardt re: Uphold logo, certainly please use their preferred logo file. Thanks!

**Comment by lgebhardt** [comment]
@alexwykoff Sorry, I forgot to link the logos [page](https://uphold.com/en/developer/api/documentation/#resources). There are quite a few options, all of which I think require a wording change.

**Comment by alexwykoff** [comment]
Ah thanks!
6.a. 
"To claim your funds you will need to create a wallet via Uphold." (https://uphold.com/en/developer/api/documentation/images/buttons/uphold-connect-white@2x.png)

**Comment by alexwykoff** [comment]
@lgebhardt a few copy updates for the screens as mentioned above:

Updated 4.b.i.
<img width="961" alt="screen shot 2017-09-18 at 4 37 36 pm" src="https://user-images.githubusercontent.com/1434040/30566489-e5a56110-9c91-11e7-8fde-f271085d8d71.png">

Updated 4.b.ii.
<img width="966" alt="screen shot 2017-09-18 at 4 37 29 pm" src="https://user-images.githubusercontent.com/1434040/30566518-fc4ed32e-9c91-11e7-93b9-13fa95d40d48.png">


Updated 4.e.
<img width="961" alt="screen shot 2017-09-18 at 4 51 47 pm" src="https://user-images.githubusercontent.com/1434040/30566537-0ddc82f8-9c92-11e7-8935-feb4c39bd143.png">

**Comment by alexwykoff** [comment]
Update : 

We'll want to separate the notion of an error condition versus waiting for notification.

To that end, if the user is waiting, they should see Screen 5.

If there is an error with verification, the user should be returned to the screen just before verification was initiated. They should be given a notification that an error has occurred. It can auto-dismiss or be closed by the user.

Screen 5.e.
<img width="962" alt="screen shot 2017-09-20 at 3 36 25 pm" src="https://user-images.githubusercontent.com/1434040/30666391-ca1d4a8c-9e19-11e7-8dd3-d8bfa9e88188.png">

**Comment by alexwykoff** [comment]
Update : 

We will want to introduce a timeout on the verification wait period. 

At the end of 5 days, we will want to both send an email to the user notifying them that verification failed and send them back one step in the flow per https://github.com/brave/publishers/issues/50#issuecomment-330975019

**Comment by ayumi** [comment]
for the landing page art, it sounds like it's TBD (I'm not a fan of the spyglass baby stock photo). i'd like to suggest something with a cool lion/bat/lat/bion.

**Comment by alexwykoff** [comment]
Chatting with @lgebhardt about layout constraints, 4.b.i & 4.b.ii have a minor update. 
4.b.i
![screen shot 2017-09-22 at 2 07 43 pm](https://user-images.githubusercontent.com/1434040/30761972-c6f97028-9fa5-11e7-9953-014a423d1f74.png)


4.b.ii
![screen shot 2017-09-22 at 2 07 37 pm](https://user-images.githubusercontent.com/1434040/30761984-d34f46ea-9fa5-11e7-8343-9ce479568d29.png)

**Comment by mrose17** [comment]
The flow is done. Design changes are captured in other issues.
